From e1a88835b1a5ff70dbaca8aa2796e83f4acb533c Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Wed, 21 Oct 2020 11:45:15 +0300
Subject: [PATCH 0828/1921] arch/arm64/boot/dts/marvell: Add SMMU stream IDs to
 AP

- Add Stream IDs to XOR and SDIO DMA masters for AP80x DTS.

Change-Id: I03eb154d20e51211be865704de753974afe7362a
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/38324
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-ap80x.dtsi | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi b/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi
index 12e477f1aeb9..5191152bd504 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi
@@ -174,6 +174,7 @@
 				msi-parent = <&gic_v2m0>;
 				clocks = <&ap_clk 3>;
 				dma-coherent;
+				iommus = <&smmu 0x1>;
 			};
 
 			xor@420000 {
@@ -183,6 +184,7 @@
 				msi-parent = <&gic_v2m0>;
 				clocks = <&ap_clk 3>;
 				dma-coherent;
+				iommus = <&smmu 0x2>;
 			};
 
 			xor@440000 {
@@ -257,6 +259,7 @@
 				clock-names = "core";
 				clocks = <&ap_clk 4>;
 				dma-coherent;
+				iommus = <&smmu 0x5>;
 				marvell,xenon-phy-slow-mode;
 				status = "disabled";
 			};
-- 
2.31.1

